SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -1422,23 +1444,36 @@ void USpatialReceiver::ResolveObjectReferences(FRepLayout& RepLayout, UObject* R
 		}
 
 		FObjectReferences& ObjectReferences = It.Value();
+
 		UProperty* Property = ObjectReferences.Property;
+
 		// ParentIndex is -1 for handover properties
+		bool bIsHandover = ObjectReferences.ParentIndex == -1;
+		FRepLayoutCmd* Cmd = ObjectReferences.CmdIndex >= 0 ? &RepLayout.Cmds[ObjectReferences.CmdIndex] : nullptr;
 		FRepParentCmd* Parent = ObjectReferences.ParentIndex >= 0 ? &RepLayout.Parents[ObjectReferences.ParentIndex] : nullptr;
 
+#if ENGINE_MINOR_VERSION <= 20
+		int32 StoredDataOffset = AbsOffset;
+#else
+		int32 StoredDataOffset = Cmd->ShadowOffset;
+#endif
+
 		if (ObjectReferences.Array)
 		{
 			check(Property->IsA<UArrayProperty>());
 
-			Property->CopySingleValue(StoredData + AbsOffset, Data + AbsOffset);
+			if (!bIsHandover)
```

Comment:
The logic for this functionality is quite different now dependent on if it's handover or not, perhaps it's better to be more explicit up front and pull it into two sections of functionality.